bee-node/bee-plugin/src/event.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Types for events.
+
+pub use crate::grpc::{MessageParsedEvent, MessageRejectedEvent, ParsingFailedEvent};
+
+use std::convert::TryFrom;
+
+/// Identifier for each event type.
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
+#[repr(u8)]
+pub enum EventId {
+    /// Identifier for [`MessageParsedEvent`].
+    MessageParsed = 0,
+    /// Identifier for [`ParsingFailedEvent`].
+    ParsingFailed = 1,
+    /// Identifier for [`MessageRejectedEvent`].
+    MessageRejected = 2,
+}
+
+impl TryFrom<u8> for EventId {
```

Comment:
Is there a crate to derive this maybe ? Can get quite error prone when the enum gets big

Reply:
Alex mentioned something about the num crate. I'll check

Reply:
I think this might be what you're looking for:
https://docs.rs/num_enum/0.5.3/num_enum/